os/user/getent_cgo.go

```diff
@@ -0,0 +1,74 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris darwin
+// +build cgo,!osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"sync"
+)
+
+// Since the getpwent is not thread safe, muUsers locks access to iterateUsers for single goroutine at a time.
+var muUsers = sync.Mutex{}
+
+// iterateUsers iterates over users database via getpwent library call. If fn returns non nil error, then
+// iteration is terminated. If result from getpwent is nil, then iteration is completed and terminated.
```

Comment:
Update comment to clarify the assumptions of underlying C library call and interaction with Go: e.g. "A nil result from `getpwent` means there were no more entries, or an error occurred. Thus, if getpwent returns null, the iteration is terminated, and error, if any, is appropriately passed through.